src/main/scala/org/broadinstitute/dsde/workbench/avram/util/RestClient.scala

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.dsde.workbench.avram.util
+
+import java.net.URI
+import java.util.logging.Logger
+
+import com.softwaremill.sttp.{HttpURLConnectionBackend, Request, Uri, sttp}
+
+/**
+  * Helpful stuff for making rest calls using sttp (https://github.com/softwaremill/sttp).
+  */
+trait RestClient {
+  private val log: Logger = Logger.getLogger(getClass.getName)
+
+  implicit val sttpBackend = HttpURLConnectionBackend()
```

Comment:
This is a synchronous backend. There are many others available: https://sttp.readthedocs.io/en/latest/backends/summary.html. Suggestions welcome.

Reply:
Not 100% sure.. probably for our use case (and given GAE's restrictions) `HttpURLConnectionBackend` is appropriate.

Reply:
Well, as long as any threads are just used in the scope of the request, I think we're okay in GAE standard.

Though we might need to move from automatic to one of the other scaling types anyway. I'm finding more and more caveats with automatic, e.g. https://cloud.google.com/endpoints/docs/frameworks/known-issues#poor_performance_on_automatic_scaling_instances

Reply:
I don't really understand what "in the scope of the request" means w.r.t. threading. Does this mean request handler threads can submit tasks to a thread pool? Or does all work need to be synchronous to a request handler thread?

That doc you linked is slightly concerning.. :)

Reply:
App Engine standard allows threads so we shouldn't be afraid of them. However, App Engine standard with automatic scaling does not support _background_ threads. (I don't think it explicitly fails but it also doesn't provide CPU time for them to run and, due to the automatic scaling, we can't rely on any background threads safely completing their tasks.) By "in the scope of the request" I mean while the main request thread is alive. During that time, we are absolutely allowed to have threads running. For example, if we had 2 independent web service calls to make, we could use something other than `HttpURLConnectionBackend` to run them in parallel and collect the results to build our response; we don't have to run them synchronously.

Reply:
Ah I see - thank you.

Reply:
i'm concerned about the threading for when we need to support asynchronous calls like loading entities from a google bucket.